service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/orders/ApproveDraftOrdersController.java

```diff
@@ -0,0 +1,143 @@
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        CaseDetailsHelper.removeTemporaryFields(caseDetails, reviewDecisionFields());
+
+        caseDetails.getData().putAll(approveDraftOrdersService.getPageDisplayControls(caseData));
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        CaseDetailsHelper.removeTemporaryFields(caseDetails, reviewDecisionFields());
+
+        caseDetails.getData().putAll(approveDraftOrdersService.populateDraftOrdersData(caseData));
+
+        if (!(caseData.getCmoToReviewList() instanceof DynamicList)) {
+            // reconstruct dynamic list
+            caseDetails.getData().put("cmoToReviewList", approveDraftOrdersService.buildDynamicList(caseData));
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/validate-review-decision/mid-event")
+    public AboutToStartOrSubmitCallbackResponse validateReviewDecision(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+        Map<String, Object> data = caseDetails.getData();
+
+        List<String> errors = approveDraftOrdersService.validateDraftOrdersReviewDecision(caseData, data);
+
+        return respond(caseDetails, errors);
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+        Map<String, Object> data = caseDetails.getData();
+
+        // reset ordersToBeSent in the notifications
+        caseDetails.getData().remove("ordersToBeSent");
+
+        Element<HearingOrdersBundle> selectedOrdersBundle =
+            approveDraftOrdersService.getSelectedHearingDraftOrdersBundle(caseData);
+
+        // review cmo
+        data.putAll(approveDraftOrdersService.reviewCMO(caseData, selectedOrdersBundle));
+
+        // review C21 orders
+        approveDraftOrdersService.reviewC21Orders(caseData, data, selectedOrdersBundle);
+
+        CaseDetailsHelper.removeTemporaryFields(caseDetails, transientFields());
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/submitted")
+    public void handleSubmitted(@RequestBody CallbackRequest callbackRequest) {
+        CaseData caseDataBefore = getCaseDataBefore(callbackRequest);
+        CaseData caseData = getCaseData(callbackRequest);
+
```

Comment:
that should trigger publishEvent(new AfterSubmissionCaseDataUpdated(caseData, caseDataBefore)) to update the  summary tab (relative to next/previous hearing)

Reply:
Discussed with @joez-f, this will be added as part of the notifications ticket ([FPLA-2562](https://tools.hmcts.net/jira/browse/FPLA-2562))